### PR TITLE
TokensAPI and PoolPairsAPI - refactor

### LINKED ISSUE #698: `TokensAPI` and `PoolPairsAPI`  - refactor
#### What would you like to be added:

We can refactor it similar with how we do it on loans and auctions (via dispatch)

### PR DESCRIPTION
**Pull request by fullstackninja864**
#### What kind of PR is this?:

/kind refactor

#### Which issue(s) does this PR fixes?:

Fixes #698

#### Developer Checklist:

- [x] Read your code changes at least once
- [x] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [x] No console errors on web
- [x] Tested on Light mode and Dark mode*
- [x] Your UI implementation visually matched the rendered design*
- [x] Unit tests*
- [ ] Added e2e tests*
- [ ] Added translations*

### CODE REVIEW

File: mobile-app/app/screens/AppNavigator/screens/Balances/screens/SendScreen.tsx [hunk 1]
```diff
@@ -49,7 +49,9 @@ export function SendScreen ({
   const logger = useLogger()
   const { networkName } = useNetworkContext()
   const client = useWhaleApiClient()
-  const tokens = useTokensAPI()
+  const { address } = useWalletContext()
+  const blockCount = useSelector((state: RootState) => state.block.count) ?? 0
```
Comment (thedoublejay) [inline mobile-app/app/screens/AppNavigator/screens/Balances/screens/SendScreen.tsx:53]: Why do you need to add 0 here? If it's undefined wouldn't it be the same?

File: mobile-app/app/screens/AppNavigator/screens/Balances/BalancesScreen.tsx [hunk 2]
```diff
@@ -45,6 +43,7 @@ export function BalancesScreen ({ navigation }: Props): JSX.Element {
     isBalancesDisplayed,
     toggleDisplayBalances: onToggleDisplayBalances
   } = useDisplayBalancesContext()
+  const blockCount = useSelector((state: RootState) => state.block.count) ?? 0
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Balances/BalancesScreen.tsx:46]: we need the fallback to `0`?

File: mobile-app/app/screens/PlaygroundNavigator/sections/PlaygroundUTXO.tsx [hunk 3]
```diff
@@ -61,7 +59,7 @@ export function PlaygroundUTXO (): JSX.Element {
             <PlaygroundAction
               onPress={async () => {
                 const address = await getActiveAddress()
-                fetchTokens(whaleApiClient, address, dispatch, logger)
+                dispatch(fetchTokens({ client: whaleApiClient, address }))
```
Comment (thedoublejay) [inline mobile-app/app/screens/PlaygroundNavigator/sections/PlaygroundUTXO.tsx:62]: maybe rename this to client? so it's consistent

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: mobile-app/app/screens/PlaygroundNavigator/sections/PlaygroundUTXO.tsx (hunk 3 above)
Comment (fullstackninja864) [reply mobile-app/app/screens/PlaygroundNavigator/sections/PlaygroundUTXO.tsx:62]: Will make changes

**Review by thedoublejay — APPROVED** [review]

**Review by kyleleow — APPROVED** [review]

**Pull request merged**